harness/tests/experiment/keras/test_keras_data.py

```diff
@@ -174,7 +173,41 @@ def test_enqueuer(
     with keras._build_enqueuer(
         sequence=IdentitySequence(100),
         workers=workers,
-        use_multiprocessing=use_multiprocessing,
+        use_multiprocessing=False,
+        max_queue_size=10,
+        shard_rank=rank,
+        num_shards=size,
+        repeat=False,
+        shuffle=shuffle,
+        shuffle_seed=777,
+        prior_batches_trained=skip,
+    ) as enqueuer:
+        assert list(enqueuer.data()) == list(sampler.yield_epoch()), "first epoch was wrong"
+        assert list(enqueuer.data()) == list(sampler.yield_epoch()), "second epoch was wrong"
+        assert list(enqueuer.data()) == list(sampler.yield_epoch()), "third epoch was wrong"
+
+
+def test_enqueuer_multiprocessing() -> None:
+    """Same enqueuer test as above, but with multiprocessing enabled.
+
+    A single, arbitrary set of parameters from the above test is chosen in order to exercise the
+    keras.enqueuer._MultiprocessingEnqueuer. In this test, multiprocessing is slow on macOS, so
+    we're minimizing the number of times we execute it assuming there's no interaction between the
+    choice of threading/multiprocessing and the examples loaded by the enqueuer.
+    """
+    epoch_len = 100
+    workers = 5
```

Comment:
I'm not sure if having just 1 multiprocessing passthrough is enough to fully test the enqueuer functionality.
Looking at the actual class, it seems to me like `shuffle` , `skip` , and `rank size` are properties of the sampler rather than the actual enqueuer, so I have no problems with keeping those fixed for a multiprocessing test of the enqueuer. 
However, I do think we should at least sweep the `workers` parameter.

Reply:
> it's also kind of more attention than I want to give this problem.
I agree.
> I don't really want to sweep workers for use_multiprocessing=True because of this conditional
Ah, I was so focused on the sampler that I didn't notice that one. Then just nix the workers=0 case since that is also already tested by the threading test.

Reply:
done